visitors/http_rest_diff/spec_diff_visitor.go

```diff
@@ -210,9 +213,39 @@ func (*DefaultSpecDiffVisitorImpl) EnterOneOfs(self interface{}, ctx http_rest.S
 		return v.EnterAddedOrRemovedOneOf(self, ctx, left, right)
 	}
 
+	if oneOfsDiffer(left, right) {
+		return v.EnterChangedOneOf(self, ctx, left, right)
+	}
+
 	return Continue
 }
 
+func oneOfsDiffer(left, right *pb.OneOf) bool {
+	if left.PotentialConflict != right.PotentialConflict {
+		return true
+	}
+
+	if len(left.Options) != len(right.Options) {
+		return true
+	}
+
+	// Manually pair up the options to see if we can get things to match.
+	rightOptions := maps.Clone(right.Options)
+OUTER:
+	for _, leftOption := range left.Options {
+		for rightKey, rightOption := range rightOptions {
+			if IsSameData(leftOption, rightOption) {
+				delete(rightOptions, rightKey)
+				continue OUTER
+			}
+		}
+
+		// No match found for leftOption.
+		return true
+	}
+	return false
```

Comment:
I don't think this is right.

Consider leftOption = {A} and rightOption = {A, B}

I think you meant 
```
return len( rightOption ) == 0
```

Reply:
Isn't this covered by the length check above the loop?

Reply:
Hm, OK, you're right, each loop should decrease the length of rightOptions by one or terminate.

Reply:
Let's port to F* real quick and verify this! 😄